llama_stack/core/routing_tables/models.py

```diff
@@ -156,3 +159,67 @@ async def update_registered_models(
                     source=RegistryEntrySource.listed_from_provider,
                 )
             )
+
+    async def cleanup_ephemeral_models(self) -> None:
+        """Clean up models that should not persist across sessions."""
+        try:
+            existing_models = await self.get_all_with_type("model")
+        except (AttributeError, RuntimeError, TypeError) as e:
+            logger.debug(f"Could not retrieve models for cleanup: {e}")
+            return
+
+        for model in existing_models:
+            if model.source == RegistryEntrySource.listed_from_provider:
+                logger.debug(f"Cleaning up ephemeral provider model: {model.identifier}")
+                try:
+                    await self.unregister_object(model)
+                except (AttributeError, RuntimeError, TypeError) as e:
+                    logger.debug(f"Could not unregister model {model.identifier}: {e}")
+                continue
+
+    async def cleanup_config_models(self) -> None:
+        """Clean up models that came from configuration (run.yaml)."""
+        try:
+            existing_models = await self.get_all_with_type("model")
+        except (AttributeError, RuntimeError, TypeError) as e:
+            logger.debug(f"Could not retrieve models for cleanup: {e}")
+            return
+
+        for model in existing_models:
+            if model.source == RegistryEntrySource.from_config:
+                logger.debug(f"Cleaning up config model: {model.identifier}")
+                try:
+                    await self.unregister_object(model)
+                except (AttributeError, RuntimeError, TypeError) as e:
+                    logger.debug(f"Could not unregister model {model.identifier}: {e}")
+                continue
+
+    async def initialize(self) -> None:
```

Comment:
if we are doing this, why even persist the models into the registry at all? I am not saying that it is wrong to do so, but it seems like that is what we are going for. namely, not having a persistent registry of models. is that a correct reading?

Reply:
no I guess what you are doing is you are killing everything which wasn't pushed via config or providers. i.e., whichever was explicitly registered by the user. that makes sense. 

but that could be implemented as a single "negative" method maybe -- `remove_models_not_registered_via_api` -- or something like that.

but the question that brings up is: there is nothing sacrosanct about the "model" resource; maybe all resources should work like this.

Reply:
i'd prefer a solution where a model list api exists in front of persistent and non-persistent models. i figured doing cleanup with be a simpler step. i've a feeling there are a number of places where code accesses the persistent store.